jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py

```diff
@@ -216,36 +236,42 @@ def _set_display_unit_of_selected_dataset(self):
         surface brightness.
         """
 
-        if not self.dataset_selected or not self.aperture_selected:
+        # all cubes are in sb so we can get display unit for plugin from SB display unit
+        # this can be changed to listen specifically to changes in surface brightness
+        # from UC plugin GlobalDisplayUnitChange message, but wiill require some refactoring
+        disp_unit = self.app._get_display_unit('sb')
+
+        # this check needs to be here because 'get_display_unit' will sometimes
+        # return non surface brightness units or even None when the app is starting
+        # up. this can be removed once that is fixed (see PR #3144)
+        if disp_unit is None or not check_if_unit_is_per_solid_angle(disp_unit):
             self.display_unit = ''
             self.flux_scaling_display_unit = ''
             return
 
-        data = self.dataset.selected_dc_item
-        if isinstance(data, list):
-            data = data[0]
-        comp = data.get_component(data.main_components[0])
-        if comp.units:
-            # if data is something-per-solid-angle, its a SB unit and we should
-            # use the selected global display unit for SB
-            if check_if_unit_is_per_solid_angle(comp.units):
-                display_unit_type = 'sb'
-            else:
-                display_unit_type = 'flux'
+        self.display_unit = disp_unit
 
-            disp_unit = self.app._get_display_unit(display_unit_type)
+        # get angle componant of surface brightness
+        # note: could add 'axis=angle' when cleaning this code up to avoid repeating this
 
-            self.display_unit = disp_unit
+        display_solid_angle_unit = check_if_unit_is_per_solid_angle(disp_unit, return_unit=True)
+        if display_solid_angle_unit is not None:
+            self.display_solid_angle_unit = display_solid_angle_unit.to_string()
+        else:
+            # there should always be a solid angle, but i think this is
+            # encountered sometimes when initializing something..
+            self.display_solid_angle_unit = ''

```

Comment:
same as above - can we now access the angle unit directly instead of pulling it out of sb?

Reply:
Eventually once we refactor this plugin to respond to unit conversion messages better, this can be cleaned up. However, since were using _get_display_unit now on every unit change, there is no axis=angle so it has to be pulled out of the surface brightness. should i add axis=angle to _get_display_unit?